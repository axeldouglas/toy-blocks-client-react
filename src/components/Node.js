import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import {
  Accordion,
  AccordionSummary,
  Typography,
  AccordionDetails,
  makeStyles,
  Box,
} from "@material-ui/core";
import * as actions from "../actions/nodes";
import colors from "../constants/colors";
import Status from "./Status";
import Blocks from "./Blocks";

const Node = ({ node, expanded, toggleNodeExpanded }) => {
  const classes = useStyles();
  const dispatch = useDispatch();

  useEffect(() => {
    if (expanded) {
      dispatch(actions.getNodeBlocks(node));
    }
    //eslint-disable-next-line
  }, [expanded]);

  return (
    <Accordion
      elevation={3}
      className={classes.root}
      expanded={expanded}
      onChange={() => toggleNodeExpanded(node)}
    >
      <AccordionSummary
        className={classes.summary}
        classes={{
          expandIcon: classes.icon,
          content: classes.content,
        }}
        expandIcon={<ExpandMoreIcon />}
      >
        <Box className={classes.summaryContent}>
          <Box>
            <Typography variant="h5" className={classes.heading}>
              {node.name || "Unknown"}
            </Typography>
            <Typography
              variant="subtitle1"
              className={classes.secondaryHeading}
            >
              {node.url}
            </Typography>
          </Box>
          <Status loading={node.loading} online={node.online} />
        </Box>
      </AccordionSummary>
      <AccordionDetails className={classes.details}>
        <Blocks node={node} />
      </AccordionDetails>
    </Accordion>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    margin: "12px 0 !important",
    boxShadow: "0px 3px 6px 1px rgba(0,0,0,0.15)",
    "&:before": {
      backgroundColor: "unset",
    },
    borderRadius: "3px",
  },
  summary: {
    padding: "0 30px 0 20px",
  },
  summaryContent: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    paddingRight: 6,
  },
  icon: {
    color: colors.iconsColor,
  },
  content: {
    margin: "14px 0 !important", // Avoid change of sizing on expanded
  },
  heading: {
    fontSize: theme.typography.pxToRem(16),
    fontStyle: "normal",
    fontWeight: "normal",
    color: colors.text,
    lineHeight: "24px",
    letterSpacing: "0.44px",
    marginBottom: "4px",
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(14),
    fontStyle: "normal",
    fontWeight: "normal",
    color: colors.text,
    lineHeight: "20px",
    letterSpacing: "0.25px",
    opacity: 0.54,
  },
  details: {
    padding: "0px 14px 14px 14px",
  },
}));

Node.propTypes = {
  node: PropTypes.shape({
    url: PropTypes.string,
    online: PropTypes.bool,
    name: PropTypes.string,
    loading: PropTypes.bool,
  }).isRequired,
  expanded: PropTypes.bool,
  toggleNodeExpanded: PropTypes.func.isRequired,
};

export default Node;
