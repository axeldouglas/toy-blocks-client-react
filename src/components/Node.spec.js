import { mount } from "enzyme";
import React from "react";
import { create } from "react-test-renderer";
import Node from "./Node";

describe("<Node />", () => {
  const toggleFn = jest.fn();
  const node = {
    url: "https://thawing-springs-53971.herokuapp.com",
    online: true,
    name: "",
    loading: false,
  };

  it("should handle onChange", () => {
    const wrapper = mount(
      <Node node={node} expanded={true} toggleNodeExpanded={toggleFn} />
    );
    wrapper.find("div[role='button']").simulate("click");
    expect(toggleFn).toBeCalled();
  });

  it("should match snapshot", () => {
    const component = create(
      <Node node={node} expanded={true} toggleNodeExpanded={toggleFn} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
