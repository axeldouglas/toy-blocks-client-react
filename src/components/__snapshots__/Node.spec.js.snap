// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Node /> should match snapshot 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root makeStyles-root-1 Mui-expanded MuiAccordion-rounded MuiPaper-elevation3 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={true}
    className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-summary-2 Mui-expanded"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content makeStyles-content-5 Mui-expanded"
    >
      <div
        className="MuiBox-root MuiBox-root-24 makeStyles-summaryContent-3"
      >
        <div
          className="MuiBox-root MuiBox-root-25"
        >
          <h5
            className="MuiTypography-root makeStyles-heading-6 MuiTypography-h5"
          >
            Unknown
          </h5>
          <h6
            className="MuiTypography-root makeStyles-secondaryHeading-7 MuiTypography-subtitle1"
          >
            https://thawing-springs-53971.herokuapp.com
          </h6>
        </div>
        <div
          className="MuiBox-root MuiBox-root-28"
        >
          <span
            className="makeStyles-dot-11 makeStyles-dot-26"
          />
          <span
            className="MuiTypography-root makeStyles-text-12 makeStyles-text-27 MuiTypography-body1"
          >
            ONLINE
          </span>
        </div>
      </div>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-icon-4 Mui-expanded MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiAccordionDetails-root makeStyles-details-8"
          >
            <div
              className="MuiBox-root MuiBox-root-29 makeStyles-blockWrapper-16"
            >
              <div
                className="MuiBox-root MuiBox-root-30 makeStyles-blockContainer-20"
              >
                <p
                  className="MuiTypography-root makeStyles-blockTitle-21 MuiTypography-body1"
                >
                  005
                </p>
                <p
                  className="MuiTypography-root makeStyles-blockDescription-22 MuiTypography-body1"
                >
                  The Human
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
